### PR TITLE
Support multiple images, tags, or digests when growing image manifests

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Part of https://github.com/kubernetes-sigs/promo-tools/issues/508 and https://github.com/kubernetes-sigs/promo-tools/issues/347, continues https://github.com/kubernetes-sigs/promo-tools/pull/507:

- Extract manifest functions into a separate package
- Update references to manifest operations
- image/manifest: Simplify/remove stutters
- image/manifest: Support multiple images, tags, or digests in `Grow()`
- image/manifest: Fixup testdata paths

Signed-off-by: Stephen Augustus <foo@auggie.dev>

#### Does this PR introduce a user-facing change?

```release-note
- image/manifest: Support multiple images, tags, or digests in `Grow()`
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/assign @puerco @cpanato 
cc: @kubernetes-sigs/release-engineering @CecileRobertMichon 
/kind feature

File: cmd/kpromo/cmd/pr/pr.go [hunk 1]
```diff
@@ -140,10 +143,20 @@ func init() {
 		&promoteOpts.images,
 		"image",
 		"",
+		// TODO: Parameterize
 		[]string{""}, // default to a single empty string image filter which will promote all images matching the tag.
 		"image to promote. If not specified, all images matching the tag will be promoted",
 	)
 
+	PRCmd.PersistentFlags().StringSliceVarP(
+		&promoteOpts.images,
```
Comment (CecileRobertMichon) [inline cmd/kpromo/cmd/pr/pr.go:152]: should be &promoteOpts.digests?

Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go:152]: Eeep, good catch!

Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go]: Fixed in https://github.com/kubernetes-sigs/promo-tools/compare/846a0134b1f53b70ace027810b91cbbf4e739a67..a8b54d25f8f76249b2c0d942ae7513469cccc560.

File: cmd/kpromo/cmd/pr/pr.go [hunk 2]
```diff
@@ -140,10 +143,20 @@ func init() {
...
 		"image to promote. If not specified, all images matching the tag will be promoted",
 	)
 
+	PRCmd.PersistentFlags().StringSliceVarP(
+		&promoteOpts.digests,
+		"digests",
+		"",
+		// TODO: Parameterize
+		[]string{""}, // default to a single empty string image filter which will promote all images matching the tag.
+		"digests to promote. If not specified, all images matching the tag will be promoted",
```
Comment (CecileRobertMichon) [inline cmd/kpromo/cmd/pr/pr.go:157]: "If not specified, all images matching the tag will be promoted" -> not sure how this translates to digests but it should probably be more specific to digests.

File: cmd/kpromo/cmd/pr/pr.go [hunk 3]
```diff
@@ -140,10 +143,20 @@ func init() {
...
 		[]string{""}, // default to a single empty string image filter which will promote all images matching the tag.
 		"image to promote. If not specified, all images matching the tag will be promoted",
 	)
 
+	PRCmd.PersistentFlags().StringSliceVarP(
+		&promoteOpts.digests,
+		"digests",
+		"",
+		// TODO: Parameterize
+		[]string{""}, // default to a single empty string image filter which will promote all images matching the tag.
```
Comment (CecileRobertMichon) [inline cmd/kpromo/cmd/pr/pr.go:156]:
```suggestion
		[]string{""}, // default to a single empty string digest filter which will promote all digests matching the tag.
```

File: cmd/kpromo/cmd/pr/pr.go (hunk 2 above)
Comment (justaugustus) [reply cmd/kpromo/cmd/pr/pr.go:157]: Dropped this altogether for both flags.
As we iterate, I'll extend the CLI guidance on the flags, which is more visible than the code comments.

**Comment by puerco** [comment]
Nice, another chunk reworked!
/lgtm

**Review by cpanato — COMMENTED** [review]
wow

late /lgtm

**Pull request merged**